source/administration/legacy-upgrade.md

```diff
@@ -1,17 +1,109 @@
-  - [Upgrade Enterprise Edition to 2.2.x and earlier](https://docs.mattermost.com/administration/legacy-upgrade.html#upgrade-enterprise-edition-to-2-2-x-and-earlier)
+If you are upgrading a server that is already running a supported version, See [Upgrading Mattermost Server](../../administration/upgrade.html).
 
-## Upgrade Team Edition
+## Upgrade Team Edition from 3.0.0 and later
 
-### Upgrade Team Edition to 3.0.x
+### Important Notices
+
+1. Security related changes were made in 3.9.0 that cause any previously created team invite links, password reset links, and email verification links to no longer work. You must update any place where you have published these links.
+
+1. Security-related changes were made in 3.8.0 that require you to verify settings in the System Console before upgrading from version 3.7.0 and earlier to any version greater than 3.8.0
+
+    1. In the GENERAL section of the System Console, click **Configuration** and make sure that the **Site URL** is specified. It must not be empty. For more information about SiteURL, see `Configuration Settings <config-settings.html#site-url>`_
+    2. In the GENERAL section of the System Console, click **Logging** and make sure that the **File Log Directory** field is either empty or has a directory path only. It must not have a filename as part of the path.
+
+1. Changes were made in 3.8.0 that require a change in the proxy configuration. If you're using NGINX:
+    1. Open the NGINX configuration file as root. The file is usually ``/etc/nginx/sites-available/mattermost`` but might be different on your system.
+    2. Locate the following line:
+     `location /api/v3/users/websocket {`
+    3. Replace the  line with ``location ~ /api/v[0-9]+/(users/)?websocket$ {``.
+
+  If you are using a proxy other than NGINX, make the equivalent change to that proxy's configuration.
+
+**To upgrade your system from 3.0.0 and later**:
+
+1. Download the latest version of Team Edition server and note any compatibility procedures:
+      - Note: If public links are enabled, upgrading from `v3.3.x` and earlier will invalidate existing public links due to a security upgrade allowing admins to invalidate links by resetting a public link salt from the System Console.
+      - Note: RHEL6 and Ubuntu installations must verify the line `limit nofile 50000 50000` is included in `/etc/init/mattermost.conf` file. See the [installation guide](../../guides/administrator.html#install-guides) for your operating system for more details.
+      - Note: RHEL7 and Ubuntu installations must verify the line `LimitNOFILE=49152` is included in the `/etc/systemd/system/mattermost.service` file. See the [installation guide](../../guides/administrator.html#install-guides) for your operating system for more details.
+      1. Download [the latest version of the Mattermost Server](https://about.mattermost.com/download/). In the following command, replace `X.X.X` with the version that you want to download:
```

Comment:
the bullet points and the numbered steps here appear as a single list. I thought for a moment that 1. and 2. here was a continuation of bullet points.